src/ui/MessageBody/index.tsx

```diff
@@ -0,0 +1,131 @@
+  } = props;
+
+  const threadMessageKindKey = useThreadMessageKindKeySelector({
+    isMobile,
+  });
+  const statefulFileInfoList = useStatefulFileInfoList(message); // For MultipleFilesMessage.
+
+  const messageTypes = getUIKitMessageTypes();
+  const isOgMessageEnabledInGroupChannel = channel?.isGroupChannel() && config.groupChannel.enableOgtag;
+
+  return <>
+    {isOgMessageEnabledInGroupChannel && isOGMessage(message as UserMessage) ? (
+      <OGMessageItemBody
+        className={MESSAGE_ITEM_BODY_CLASSNAME}
+        message={message as UserMessage}
+        isByMe={isByMe}
+        mouseHover={mouseHover}
+        isMentionEnabled={config?.isMentionEnabled || false}
+        isReactionEnabled={isReactionEnabledInChannel}
+        onMessageHeightChange={onMessageHeightChange}
+      />
+    ) : isTextMessage(message as UserMessage) && (
+      <TextMessageItemBody
+        className={MESSAGE_ITEM_BODY_CLASSNAME}
+        message={message as UserMessage}
+        isByMe={isByMe}
+        mouseHover={mouseHover}
+        isMentionEnabled={config?.isMentionEnabled || false}
+        isReactionEnabled={isReactionEnabledInChannel}
+      />
+    )}
+    {(getUIKitMessageType((message as CoreMessageType)) === messageTypes.FILE) && (
+      <FileMessageItemBody
+        className={MESSAGE_ITEM_BODY_CLASSNAME}
+        message={message as FileMessage}
+        isByMe={isByMe}
+        mouseHover={mouseHover}
+        isReactionEnabled={isReactionEnabledInChannel}
+      />
+    )}
+    {isMultipleFilesMessage(message as CoreMessageType) && (
+      <MultipleFilesMessageItemBody
+        className={MESSAGE_ITEM_BODY_CLASSNAME}
+        message={message as MultipleFilesMessage}
+        isByMe={isByMe}
+        mouseHover={mouseHover}
+        isReactionEnabled={isReactionEnabledInChannel}
+        threadMessageKindKey={threadMessageKindKey}
+        statefulFileInfoList={statefulFileInfoList}
+      />
+    )}
+    {isVoiceMessage(message as FileMessage) && (
+      <VoiceMessageItemBody
+        className={MESSAGE_ITEM_BODY_CLASSNAME}
+        message={message as FileMessage}
+        channelUrl={channel?.url ?? ''}
+        isByMe={isByMe}
+        isReactionEnabled={isReactionEnabledInChannel}
+      />
+    )}
+    {(isThumbnailMessage(message as FileMessage)) && (
+      <ThumbnailMessageItemBody
+        className={MESSAGE_ITEM_BODY_CLASSNAME}
+        message={message as FileMessage}
+        isByMe={isByMe}
+        mouseHover={mouseHover}
+        isReactionEnabled={isReactionEnabledInChannel}
+        showFileViewer={showFileViewer}
+        style={isMobile ? {width: '100%'} : {}}
+      />
+    )}
+    {(getUIKitMessageType((message as CoreMessageType)) === messageTypes.UNKNOWN) && (
+      <UnknownMessageItemBody
+        className={MESSAGE_ITEM_BODY_CLASSNAME}
+        message={message}
+        isByMe={isByMe}
+        mouseHover={mouseHover}
+        isReactionEnabled={isReactionEnabledInChannel}
+      />
+    )}
+  </>
+}
```

Comment:
The type of the message to be rendered is determined unconditionally by one, so I believe it is more appropriate to handle branching rather than attempting to render all components with the && operator.

Reply:
let me apply this change and then ping you

Reply:
done @bang9